conans/client/toolchain/make.py

```diff
@@ -0,0 +1,229 @@
+# coding=utf-8
+
+import platform
+import textwrap
+
+from jinja2 import Template
+
+from conans.client.build.compiler_flags import architecture_flag, build_type_flags
+from conans.client.build.cppstd_flags import cppstd_from_settings, cppstd_flag_new as cppstd_flag
+from conans.client.tools.oss import cross_building, \
+    detected_architecture, detected_os, get_gnu_triplet, get_target_os_arch, get_build_os_arch
+from conans.errors import ConanException
+from conans.util.files import save
+
+
+class MakeToolchain(object):
+    filename = "conan_toolchain.mak"
+
+    _template_toolchain = textwrap.dedent("""
+        # Conan generated toolchain file
+        ifndef CONAN_TOOLCHAIN_INCLUDED
+            CONAN_TOOLCHAIN_INCLUDED = TRUE
+            CONAN_TC_BUILD_TYPE = {{build_type}}
+            CONAN_TC_OS_HOST = {{os_host}}
+            CONAN_TC_ARCH_HOST = {{arch_host}}
+            CONAN_TC_TRIPLET_HOST = {{triplet_host}}
+            CONAN_TC_OS_BUILD = {{os_build}}
+            CONAN_TC_ARCH_BUILD = {{arch_build}}
+            CONAN_TC_TRIPLET_BUILD = {{triplet_build}}
+            CONAN_TC_OS_TARGET = {{os_target}}
+            CONAN_TC_ARCH_TARGET = {{arch_target}}
+            CONAN_TC_TRIPLET_TARGET = {{triplet_target}}
+            CONAN_TC_COMPILER = {{compiler}}
+            CONAN_TC_COMPILER_VERSION = {{compiler_version}}
+            CONAN_TC_COMPILER_RUNTIME = {{compiler_runtime}}
+            CONAN_TC_LIBCXX = {{libcxx}}
+            CONAN_TC_CPPSTD_FLAG = {{cppstd_flag}}
+            CONAN_TC_ARCH_FLAG = {{arch_flag}}
+            CONAN_TC_BUILD_TYPE_FLAGS = {{build_type_flags}}
+            CONAN_TC_DEFINES = {{defines}}
+
+            CONAN_TC_SET_LIBCXX = {{set_libcxx}}
+            CONAN_TC_SET_CPPSTD = {{set_cppstd}}
+            CONAN_TC_SET_ARCH = {{set_arch}}
+            CONAN_TC_SET_FPIC = {{set_fpic}}
+            CONAN_TC_SET_SHARED = {{set_shared}}
+
+            CONAN_TC_CFLAGS += $(CONAN_TC_BUILD_TYPE_FLAGS)
+            CONAN_TC_CXXFLAGS += $(CONAN_TC_BUILD_TYPE_FLAGS)
+
+            ifeq ($(CONAN_TC_BUILD_TYPE),Release)
+                CONAN_TC_DEFINES += NDEBUG
+            endif
+
+            ifeq ($(CONAN_TC_SET_LIBCXX),True)
+                CONAN_TC_CLANG_BASED := $(if $(filter $(CONAN_TC_COMPILER),clang apple-clang),true)
+                ifeq ($(CONAN_TC_CLANG_BASED),True)
+                    CONAN_TC_LIBSTDCXX_BASED := $(if $(filter $(CONAN_TC_LIBCXX),libstdc++ libstdc++11),true)
+                    ifeq ($(CONAN_TC_LIBSTDCXX_BASED),True)
+                        CONAN_TC_CXXFLAGS += -stdlib=libstdc++
+                    else ifeq ($(CONAN_TC_LIBCXX),libc++)
+                        CONAN_TC_CXXFLAGS += -stdlib=libc++
+                    endif
+                else ifeq ($(CONAN_TC_COMPILER),sun-cc)
+                    ifeq ($(CONAN_TC_LIBCXX),libCstd)
+                        CONAN_TC_CXXFLAGS += -library=Cstd++
```

Comment:
Something similar to the mapping from the CONAN_TC_LIBCXX to the final flag to be passed. It is likely that this is already somewhere in python. So maybe the final toolchain file could be something like just a single line defining the ``-library=xxx`` value, but without conditions at all?

Reply:
Same answer as above. Copied patterns from Cmake. Indeed, if we can release this first iteration, we can refactor the common code out together as one concerted initiative.